veracruz-server/src/veracruz_server_linux.rs

```diff
@@ -263,11 +271,19 @@ pub mod veracruz_server_linux {
                 VeracruzServerError::VeracruzUtilError(e)
             })?;
 
-            let runtime_enclave_binary_path = env::var("RUNTIME_ENCLAVE_BINARY_PATH")
-                .unwrap_or(RUNTIME_ENCLAVE_BINARY_PATH.to_string());
+            // temporary directory to store image
+            let runtime_enclave_binary_dir = tempfile::tempdir()?;
+
+            let runtime_enclave_binary_path = runtime_enclave_binary_dir.path().join("runtime_enclave_binary");
+            fs::write(&runtime_enclave_binary_path, RUNTIME_ENCLAVE_BINARY_IMAGE)?;
+
+            // make sure our image is executable
+            let mut runtime_enclave_binary_permissions = fs::metadata(&runtime_enclave_binary_path)?.permissions();
+            runtime_enclave_binary_permissions.set_mode(0o544);
```

Comment:
Why is it `0o544`? Doesn't that mean readable but not executable by other people? Or is this different from `chmod`?

Reply:
No specific reason, I just needed to set the executable bit really (yep this is basically `chmod`). It's a temporary file that just needs to be executable and readable, so I figured to keep the permissions conservative, but there isn't any real security benefit to it.

Reply:
So `0o500` might work just as well? If so, it's perhaps worth doing that instead and adding an explanatory comment like `// readable and executable by user`.

Reply:
Sounds reasonable to me, will do